### PR TITLE
Add memc support for STM32F7 series

### PR DESCRIPTION
**Pull request by papadkostas**
As per #36204 and #29686, adding support for STM32F7 and board STM32F769I-DISC0.

### CODE REVIEW

**Comment by erwango** [comment]
@papadkostas Thanks for this change, can you check build errors reported by CI: https://buildkite.com/zephyr/zephyr/builds/47049 ?

**Comment by papadkostas** [comment]
Ohh, I'll have a look on the errors. I did another commit on the same branch of this PR for the support for the touch screen. At first there were no such errors I think. I'll revert it and check again. Anyway, I'll be back in few hours!

**Comment by papadkostas** [comment]
@erwango I think now it's good.

File: boards/arm/stm32f769i_disco/stm32f769i_disco.dts [hunk 1]
```diff
@@ -160,3 +165,45 @@ arduino_serial: &usart6 {};
...
+			 &fmc_a8_pf14 &fmc_a9_pf15 &fmc_a10_pg0 &fmc_a11_pg1
+			 &fmc_a12_pg2 &fmc_a14_pg4 &fmc_a15_pg5
+			 &fmc_d0_pd14 &fmc_d1_pd15 &fmc_d2_pd0 &fmc_d3_pd1
+			 &fmc_d4_pe7 &fmc_d5_pe8 &fmc_d6_pe9 &fmc_d7_pe10
+			 &fmc_d8_pe11 &fmc_d9_pe12 &fmc_d10_pe13 &fmc_d11_pe14
+			 &fmc_d12_pe15 &fmc_d13_pd8 &fmc_d14_pd9 &fmc_d15_pd10
+			 &fmc_d16_ph8 &fmc_d17_ph9 &fmc_d18_ph10 &fmc_d19_ph11
+			 &fmc_d20_ph12 &fmc_d21_ph13 &fmc_d22_ph14 &fmc_d23_ph15
+			 &fmc_d24_pi0 &fmc_d25_pi1 &fmc_d26_pi2 &fmc_d27_pi3 
+			 &fmc_d28_pi6 &fmc_d29_pi7 &fmc_d30_pi9 &fmc_d31_pi10>;
```
Comment (gmarull) [inline boards/arm/stm32f769i_disco/stm32f769i_disco.dts:185]: ditto

**Review by gmarull — CHANGES_REQUESTED** [review]

File: boards/arm/stm32f769i_disco/stm32f769i_disco.dts [hunk 2]
```diff
@@ -22,6 +22,11 @@
 		zephyr,flash-controller = &mx25l51245g;
 	};
 
+	sdram1: sdram@c0000000 {
+		device_type = "memory";
+		reg = <0xc0000000 DT_SIZE_M(16)>;
```
Comment (erwango) [inline boards/arm/stm32f769i_disco/stm32f769i_disco.dts:27]: According to board manual SDRAM embedded is 128M, any reason to set 16M only ?

**Review by erwango — COMMENTED** [review]
One question, otherwise sounds good except the points already reported by @gmarull

File: boards/arm/stm32f769i_disco/stm32f769i_disco.dts (hunk 2 above)
Comment (explora26) [reply boards/arm/stm32f769i_disco/stm32f769i_disco.dts:27]: 128-Mbit = 16M bytes, right ?

Comment (erwango) [reply boards/arm/stm32f769i_disco/stm32f769i_disco.dts:27]: Right :-), too early review..

File: dts/arm/st/f7/stm32f7.dtsi [hunk 3]
```diff
@@ -79,14 +79,14 @@
 			reg = <0xa0000000 0x400>;
 			clocks = <&rcc STM32_CLOCK_BUS_AHB3 0x00000001>;
 			label = "STM32_FMC";
-			status = "okay";
```
Comment (erwango) [inline dts/arm/st/f7/stm32f7.dtsi]: That's a good catch, but please fix the initial commit.

File: boards/arm/stm32f769i_disco/stm32f769i_disco.dts [hunk 4]
```diff
@@ -170,27 +170,27 @@ arduino_serial: &usart6 {};
 	status = "okay";
 	pinctrl-0 = <&fmc_nbl0_pe0 &fmc_nbl1_pe1 &fmc_nbl2_pi4 &fmc_nbl3_pi5
 		     &fmc_sdclk_pg8 &fmc_sdnwe_ph5 &fmc_sdcke0_ph2 &fmc_sdne0_ph3 
-			 &fmc_sdnras_pf11 &fmc_sdncas_pg15
```
Comment (erwango) [inline boards/arm/stm32f769i_disco/stm32f769i_disco.dts]: Please fix directly the initial commit

File: boards/arm/stm32f769i_disco/stm32f769i_disco.dts [hunk 5]
```diff
@@ -160,3 +165,45 @@ arduino_serial: &usart6 {};
...
+					    STM32_FMC_SDRAM_MWID_32
+					    STM32_FMC_SDRAM_NB_4
+					    STM32_FMC_SDRAM_CAS_3
+					    STM32_FMC_SDRAM_SDCLK_PERIOD_2
+					    STM32_FMC_SDRAM_RBURST_ENABLE
+					    STM32_FMC_SDRAM_RPIPE_0>;
+			st,sdram-timing = <2 6 4 6 2 2 2>;
+		};
+	};
+};
```
Comment (gmarull) [inline boards/arm/stm32f769i_disco/stm32f769i_disco.dts:209]: nit: missing blank line

**Review by erwango — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by ABOSTM — APPROVED** [review]

**Comment by erwango** [comment]
@papadkostas Can you fix the compliance test ?

File: boards/arm/stm32f769i_disco/stm32f769i_disco.dts [hunk 6]
```diff
@@ -160,3 +165,45 @@ arduino_serial: &usart6 {};
...
+		     &fmc_a8_pf14 &fmc_a9_pf15 &fmc_a10_pg0 &fmc_a11_pg1
+		     &fmc_a12_pg2 &fmc_a14_pg4 &fmc_a15_pg5
+		     &fmc_d0_pd14 &fmc_d1_pd15 &fmc_d2_pd0 &fmc_d3_pd1
+		     &fmc_d4_pe7 &fmc_d5_pe8 &fmc_d6_pe9 &fmc_d7_pe10
+		     &fmc_d8_pe11 &fmc_d9_pe12 &fmc_d10_pe13 &fmc_d11_pe14
+		     &fmc_d12_pe15 &fmc_d13_pd8 &fmc_d14_pd9 &fmc_d15_pd10
+		     &fmc_d16_ph8 &fmc_d17_ph9 &fmc_d18_ph10 &fmc_d19_ph11
+		     &fmc_d20_ph12 &fmc_d21_ph13 &fmc_d22_ph14 &fmc_d23_ph15
+		     &fmc_d24_pi0 &fmc_d25_pi1 &fmc_d26_pi2 &fmc_d27_pi3
+		     &fmc_d28_pi6 &fmc_d29_pi7 &fmc_d30_pi9 &fmc_d31_pi10>;
```
Comment (gmarull) [inline boards/arm/stm32f769i_disco/stm32f769i_disco.dts:185]: missing: `pinctrl-names = "default"`

**Review by gmarull — CHANGES_REQUESTED** [review]
please, fix author/signed-off compliance warnings, add missing pinctrl-names

**Comment by papadkostas** [comment]
> please, fix author/signed-off compliance warnings, add missing pinctrl-names

I fixed them, but the compliance failure that appeared now, I don't know how to fix.
58b3f6f4b9e1bb590eba4289565ad49b6c6ea0ff is probably a commit in the past but still appearing somewhere, I'm not quite sure how to deal with it. I didn't read carefully the contribution guidelines at first to avoid all these problems.
I'll make a new PR. Excuse me! :(

**Pull request closed without merging**